tests/test_interpreters.py

```diff
@@ -0,0 +1,134 @@
+from __future__ import annotations
```

Comment:
Could you please rename this to

test_multiple_interpreters.py

(and the two new mod_*.cpp files, too)?